src/main/java/apoc/util/Util.java

```diff
@@ -528,7 +528,7 @@ public static String cleanUrl(String url) {
             if (source.getRef() != null) file += "#"+source.getRef();
             return new URL(source.getProtocol(),source.getHost(),source.getPort(),file).toString();
         } catch (MalformedURLException mfu) {
-            return "invalid URL";
+            return String.format("invalid URL (%s)", url);
```

Comment:
are we sure this doesn't expose credentials?

Reply:
This method is (at this moment) used in these classes:
- LoadCsv
- LoadXls
- Xml
- JsonUtil
So we have no DB connector that could expose credentials, the only credentials that could be exposed are the http embedded which are deprecated. [Here](https://medium.com/@lmakarov/say-goodbye-to-urls-with-embedded-credentials-b051f6c7b6a3) an article that explains all.
If you think we must prevent this case we'll revert it.